### PR TITLE
Add role list command with no argument

### PR DESCRIPTION
**Pull request by nhawke**
Using the command '!role list' with no role argument will list all the 
roles that can be managed by memebot.

The bot can manage all roles below its highest role in the hierarchy.

### CODE REVIEW

File: commands.py [hunk 1]
```diff
@@ -173,7 +179,22 @@ async def role(args: List[str]) -> 'CommandOutput':
             if role.name == target_name:
                 target_role = role
                 break
-
+                    
+        # handle list with no role name
+        if action == 'list' and target_role is None:
```
Comment (super-cooper) [inline commands.py:184]: Could this not just be moved into the existing list command?

Comment (super-cooper) [reply commands.py:184]: Could we also print out an extra message for "role not found" to differentiate `!role list` and `!role list invalid-role`?

For example, for `!role list invalid-role` we could just print "Could not find role `invalid-role`" and then run the same command we currently do.

File: commands.py [hunk 2]
```diff
@@ -173,7 +179,22 @@ async def role(args: List[str]) -> 'CommandOutput':
...
+            roles = []
+            can_manage = False
+            # Bot can manage all roles below its highest role. Find that role 
+            # and begin listing roles from that point onward.
+            for role in guild.roles[:0:-1]:  # top-down, excluding @everyone
+                if can_manage:
+                    roles.append(role.name)
+                    continue
+                if Commands.client.user in role.members:
```
Comment (super-cooper) [inline commands.py:193]: We can get rid of `continue` and make the below line an `elif`.

Comment (nhawke) [reply commands.py]: done

File: commands.py (hunk 1 above)
Comment (nhawke) [reply commands.py:184]: There is a later check for `target_role == None` that is the reason this logic is earlier than the other `list` logic and returns an error message saying the role wasn't found.

Comment (super-cooper) [reply commands.py:184]: I would prefer the same subcommand to be together.

Based on my testing, it does not
image

Comment (nhawke) [reply commands.py]: whoops. the check should be against `target_name` not `target_role`. Fixed.

**Pull request merged**